DllImport by ordinal does not work

<!--
If you are new to the project get yourself familiar with https://www.mono-project.com/community/bugs/make-a-good-bug-report/ before filling the issue
-->

## Steps to Reproduce

1. Compile test case:
```
using System;
using System.Runtime.InteropServices;

static class TestOrdinalImport
{
	[DllImport("shlwapi", EntryPoint="#33")]
	unsafe extern static bool IsCharDigitW(char ch);

    public static int Main()
    {
		if (!IsCharDigitW('2'))
			return 1;
		if (IsCharDigitW('s'))
			return 2;
		return 0;
    }
}
```
2. Run on a Windows build of Mono.

<!--
You may drag & drop the attachment (repro code/solution, screenshot, etc.) onto the issue.
-->

### Current Behavior

Fails with EntryPointNotFoundException

### Expected Behavior

Exits successfully.

## On which platforms did you notice this

[ ] macOS
[ ] Linux
[x] Windows

**Version Used**:

Mono JIT compiler version 6.12.0 (Visual Studio built mono)

## Stacktrace

```
Unhandled Exception:
System.EntryPointNotFoundException: #33 assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) TestOrdinalImport.IsCharDigitW(char)
  at TestOrdinalImport.Main () [0x00001] in <8a2d827641004700b33ccad83e37b6ce>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.EntryPointNotFoundException: #33 assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) TestOrdinalImport.IsCharDigitW(char)
  at TestOrdinalImport.Main () [0x00001] in <8a2d827641004700b33ccad83e37b6ce>:0
```

--- a/mono/metadata/native-library.c
+++ b/mono/metadata/native-library.c
@@ -1456,16 +1456,6 @@ pinvoke_probe_for_symbol (MonoDl *module, MonoMethodPInvoke *piinfo, const char 
 
 	g_assert (error_msg_out);
 
-#ifdef HOST_WIN32
-	if (import && import [0] == '#' && isdigit (import [1])) {
-		char *end;
-		long id;
-
-		id = strtol (import + 1, &end, 10);
-		if (id > 0 && *end == '\0')
-			import++;
-	}
-#endif
 	mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_DLLIMPORT,
 				"Searching for '%s'.", import);
 

--- a/mono/utils/mono-dl-windows.c
+++ b/mono/utils/mono-dl-windows.c
@@ -157,7 +157,21 @@ mono_dl_lookup_symbol (MonoDl *module, const char *symbol_name)
 
 	/* get the symbol directly from the specified module */
 	if (!module->main_module)
+	{
+		if (symbol_name[0] == '#')
+		{
+			/* lookup by ordinal */
+			unsigned long ord;
+			char *end;
+
+			ord = strtoul(symbol_name + 1, &end, 10);
+
+			if (*end == '\0' && ord > 0 && ord < 65536)
+				symbol_name = (const char*)(uintptr_t)ord;
+		}
+
 		return (void*)GetProcAddress ((HMODULE)module->handle, symbol_name);
+	}
 
 	/* get the symbol from the main module */
 	proc = (gpointer)GetProcAddress ((HMODULE)module->handle, symbol_name);

